modules/core/include/visp3/core/vpImageMorphology.h

```diff
@@ -78,15 +78,60 @@ class VISP_EXPORT vpImageMorphology
                     diagonal) */
   } vpConnexityType;
 
+private:
+  /**
+   * @brief Modify the image by applying the \b operation on each of its elements on a 3x3
+   * grid.
+   * 
+   * @tparam T Either a class such as vpRGBa or a type such as double, unsigned char ...
+   * @param I The image we want to modify.
+   * @param null_value The value that is padded to the input image to manage the borders.
+   * @param operation The operation to apply to its elements on a 3x3 grid.
+   * @param connexity Either a 4-connexity, if we want to take into account only the horizontal
+   * and vertical neighbors, or a 8-connexity, if we want to also take into account the diagonal neighbors.
+   */
+  template <typename T>
+  static void imageOperation(vpImage<T> &I, const T &null_value, const T& (*operation)(const T&, const T&), const vpConnexityType &connexity = CONNEXITY_4);
+
 public:
   template <class Type>
   static void erosion(vpImage<Type> &I, Type value, Type value_out, vpConnexityType connexity = CONNEXITY_4);
 
   template <class Type>
   static void dilatation(vpImage<Type> &I, Type value, Type value_out, vpConnexityType connexity = CONNEXITY_4);
 
-  static void erosion(vpImage<unsigned char> &I, const vpConnexityType &connexity = CONNEXITY_4);
-  static void dilatation(vpImage<unsigned char> &I, const vpConnexityType &connexity = CONNEXITY_4);
+  template <typename T>
+  static void erosion(vpImage<T> &I, const vpConnexityType &connexity = CONNEXITY_4);
+
+  template <typename T>
+  static void dilatation(vpImage<T> &I, const vpConnexityType &connexity = CONNEXITY_4);
+
+  #if defined(VISP_BUILD_DEPRECATED_FUNCTIONS)
+  /*!
+    @name Deprecated functions
+  */
+  //@{
+  /*!
+     \deprecated Provided only for compat with previous releases. Use rather the template function erosion(vpImage<T> &, const vpConnexityType &)
+     
+     Add here the description of the function...
+   */
+  vp_deprecated static void erosion(vpImage<unsigned char> &I, const vpConnexityType &connexity = CONNEXITY_4) 
+  {
+    vpImageMorphology::erosion<unsigned char>(I, connexity);
+  }
+
+  /*!
+     \deprecated Provided only for compat with previous releases. Use rather the template function dilatation(vpImage<T> &, const vpConnexityType &)
+     
+     Add here the description of the function...
```

Comment:
Should be updated

Reply:
Sorry, it has been corrected in commit af0e026